upup/pkg/fi/utils/sanitize_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"testing"
+
+	"k8s.io/client-go/util/homedir"
+)
+
+func TestSanitizeString(t *testing.T) {
+	tests := []struct {
+		s        string
+		expected string
+	}{
+		{
+			s:        "test",
+			expected: "test",
+		},
+		{
+			s:        "/%test!",
+			expected: "__test_",
+		},
+	}
+
+	for _, test := range tests {
+		result := SanitizeString(test.s)
+		if test.expected != result {
+			t.Errorf("Expected %s, got %s", test.expected, result)
+		}
+	}
+}
+
+func TestExpandPath(t *testing.T) {
+	tests := []struct {
+		s        string
+		expected string
+	}{
+		{
+			s:        "test",
+			expected: "test",
+		},
+		{
+			s:        "~/test",
+			expected: homedir.HomeDir() + "/test",
```

Comment:
This test is pretty much the implementation of `ExpandPath`. I would change this to a hardcoded string.

Reply:
Because the HomeDir obtained in different test environments is different, I think it is impossible to write a definite string.

Reply:
The only relevant environment is linux. you can skip if the environment is something else. If not this test is rather meaningless.

Reply:
Getting the value of `homedir.HomeDir ()` in a linux environment is also different, so I will delete this test. Thanks!